ArduCopter/fence.cpp

```diff
@@ -63,7 +63,7 @@ void Copter::fence_check()
 
             // disarm immediately if we think we are on the ground or in a manual flight mode with zero throttle
             // don't disarm if the high-altitude fence has been broken because it's likely the user has pulled their throttle to zero to bring it down
-            if (ap.land_complete || (flightmode->has_manual_throttle() && ap.throttle_zero && !failsafe.radio && ((fence.get_breaches() & AC_FENCE_TYPE_ALT_MAX)== 0))){
+            if (ap.land_complete || (flightmode->has_manual_throttle() && ap.throttle_zero && !failsafe.radio && ((fence.get_breaches() & (AC_FENCE_TYPE_ALT_MAX | AC_FENCE_TYPE_ALT_MIN))== 0))){
```

Comment:
Isn't this logic already screwed?  If you're breaching both a polygon fence and the max-alt fence then we won't do this disarm thing, which doesn't seem right.